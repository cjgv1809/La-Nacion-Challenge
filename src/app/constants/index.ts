const sectionsLinks = [
  {
    href: "https://www.lanacion.com.ar/ultimas-noticias/",
    text: "Últimas noticias",
  },
  { href: "https://www.lanacion.com.ar/economia/", text: "Economía" },
  { href: "https://www.lanacion.com.ar/sociedad/", text: "Sociedad" },
  { href: "https://www.lanacion.com.ar/deportes/", text: "Deportes" },
  { href: "https://www.lanacion.com.ar/espectaculos/", text: "Espectáculos" },
  { href: "https://lnmas.lanacion.com.ar/", text: "LN+" },
];

const politicsLinks = [
  { href: "https://www.lanacion.com.ar/politica/", text: "Política" },
  { href: "https://www.lanacion.com.ar/el-mundo/", text: "El mundo" },
  { href: "https://www.lanacion.com.ar/opinion/", text: "Opinión" },
  { href: "https://www.lanacion.com.ar/lifestyle/", text: "Lifestyle" },
  {
    href: "https://edicionimpresa.lanacion.com.ar/la-nacion",
    text: "Edición impresa",
  },
  { href: "https://club.lanacion.com.ar/", text: "Club LA NACION" },
];

const magazinesLinks = [
  { href: "https://www.somosohlala.com/", text: "OHLALÁ!" },
  { href: "https://www.lanacion.com.ar/revista-hola/", text: "¡HOLA!" },
  { href: "https://www.lanacion.com.ar/revista-living/", text: "LIVING" },
  { href: "https://www.lanacion.com.ar/revista-jardin/", text: "JARDÍN" },
  { href: "https://www.lanacion.com.ar/revista-lugares/", text: "LUGARES" },
  { href: "https://es.rollingstone.com/arg/", text: "ROLLING STONE" },
];

const bonvivirLinks = [
  {
    href: "https://bonvivir.com/?_gl=1*hmp4mf*_gcl_aw*R0NMLjE3MzAyMjgwMTYuQ2owS0NRandqNEs1QmhEWUFSSXNBRDFMeTJxOHQxaWhQTjc5dTBYT2x3R3NoajRuZUdrMDdBNko3QWpBMWZiMVgwaUJkOWxkc2h6emluUWFBdks1RUFMd193Y0I.*_gcl_au*MTcyNjc3MDM2LjE3MzAyMjgwMTU.",
    text: "Bonvivir",
  },
  {
    href: "https://colecciones.lanacion.com.ar/",
    text: "Colecciones",
  },
  {
    href: "https://www.utdt.edu/ver_contenido.php?id_contenido=1111&id_item_menu=2327",
    text: "Master en Periodismo",
  },
  {
    href: "https://fundacionlanacion.org.ar/?_gl=1*14zmtm4*_gcl_aw*R0NMLjE3MzAyMjgwMTYuQ2owS0NRandqNEs1QmhEWUFSSXNBRDFMeTJxOHQxaWhQTjc5dTBYT2x3R3NoajRuZUdrMDdBNko3QWpBMWZiMVgwaUJkOWxkc2h6emluUWFBdks1RUFMd193Y0I.*_gcl_au*MTcyNjc3MDM2LjE3MzAyMjgwMTU.",
    text: "Fundación LA NACION",
  },
];

const generalLinks = [
  {
    href: "https://www.lanacion.com.ar/mapa-del-sitio/",
    text: "Mapa del sitio",
  },
  { href: "https://www.contacto.lanacion.com.ar/ayuda", text: "Ayuda" },
  { href: "https://club.lanacion.com.ar/ayuda/", text: "Atención al socio" },
  {
    href: "https://www.contacto.lanacion.com.ar/tyc",
    text: "Términos y condiciones",
  },
  { href: "https://www.lanacion.in/", text: "¿Cómo anunciar?" },
  {
    href: "https://suscripciones.lanacion.com.ar/suscribirme",
    text: "Suscribirse al diario impreso",
  },
];

export {
  sectionsLinks,
  politicsLinks,
  magazinesLinks,
  bonvivirLinks,
  generalLinks,
};
