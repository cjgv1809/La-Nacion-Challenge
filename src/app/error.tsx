"use client";

import { useRouter } from "next/navigation";
import Button from "@/app/components/common/Button";

interface ErrorPageProps {
  error: Error;
}

export default function ErrorPage({ error }: ErrorPageProps) {
  const router = useRouter();

  return (
    <section className="error-container">
      <div className="error-container__card">
        <h1>Ups! algo salió mal</h1>
        <p>{error.message}</p>
        <Button
          text="Ir a la página principal"
          className="--btn --primary"
          onClick={() => router.push("/")}
        />
      </div>
    </section>
  );
}
