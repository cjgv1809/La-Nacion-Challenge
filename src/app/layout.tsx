import type { Metadata } from "next";
import Header from "./components/layout/Header/Header";
import Footer from "./components/layout/Footer/Footer";
import { config } from "@fortawesome/fontawesome-svg-core";
import "@fortawesome/fontawesome-svg-core/styles.css";
import "./globals.css";

// Prevent Font Awesome from adding its CSS since we're doing it
config.autoAddCss = false;

export const metadata: Metadata = {
  title: "La Nacion - Challenge FE",
  description: "La Nacion Challenge Frontend",
  keywords: ["La Nacion", "Challenge", "FE", "2024"],
  authors: [{ name: "Carlos Gomes" }],
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="es-AR">
      <head>
        <link
          rel="preload"
          href="/@fortawesome/fontawesome-svg-core/styles.css"
          as="style"
        />
      </head>
      <body>
        <Header />
        {children}
        <Footer />
      </body>
    </html>
  );
}
