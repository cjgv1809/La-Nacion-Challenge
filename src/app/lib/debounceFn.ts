function debounceFn(func: (...args: unknown[]) => void, delay: number) {
  let timeoutId: NodeJS.Timeout;
  return function (...args: unknown[]) {
    clearTimeout(timeoutId);
    timeoutId = setTimeout(() => func(...args), delay);
  };
}

export default debounceFn;
