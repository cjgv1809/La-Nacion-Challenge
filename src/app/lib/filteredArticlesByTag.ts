import type { Article } from "../types";

const filteredArticlesByTag = (
  articles: Article[],
  tagSlug: string
): Article[] =>
  articles.filter((article) =>
    article.taxonomy.tags.some((tag) => tag.slug === tagSlug)
  );

export default filteredArticlesByTag;
