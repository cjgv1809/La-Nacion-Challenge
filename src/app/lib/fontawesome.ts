import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faFacebook,
  faXTwitter,
  faInstagram,
} from "@fortawesome/free-brands-svg-icons";
import { faRss } from "@fortawesome/free-solid-svg-icons";

// Initialize icon library with only the icons we need
library.add(faFacebook, faXTwitter, faInstagram, faRss);
