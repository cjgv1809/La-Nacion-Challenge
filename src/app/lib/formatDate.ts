const formatDate = (date: Date): string =>
  new Date(date).toLocaleDateString("es-AR", {
    day: "numeric",
    month: "long",
    year: "numeric",
  });

export default formatDate;
