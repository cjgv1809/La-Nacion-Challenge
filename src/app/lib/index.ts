import getTagsData from "./getTagsData";
import getArticlesData from "./getArticlesData";
import formatDate from "./formatDate";

export { getTagsData, getArticlesData, formatDate };
