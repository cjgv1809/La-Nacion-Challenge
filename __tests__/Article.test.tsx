import { render, screen } from "@testing-library/react";
import ArticleComponent from "@/app/components/Article";
import { formatDate } from "@/app/lib";

// render a fixed date for testing
const newDate = new Date("2024-01-01T00:00:00Z"); // January 1, 2024
const formattedDate = formatDate(newDate);

const article = {
  _id: Math.random().toString(36).substring(7),
  promo_items: {
    basic: {
      url: "https://via.placeholder.com/800x400",
    },
  },
  headlines: {
    basic: "Article headline",
  },
  display_date: newDate,
};

describe("Article", () => {
  it("renders the article", () => {
    render(<ArticleComponent article={article} />);

    const articleElement = screen.getByRole("article");
    expect(articleElement).toBeInTheDocument();
  });

  it("renders the article headline", () => {
    render(<ArticleComponent article={article} />);

    const articleHeadline = screen.getByRole("heading");
    expect(articleHeadline).toHaveTextContent(article.headlines.basic);
  });

  it("renders the article image", () => {
    render(<ArticleComponent article={article} />);

    const articleImage = screen.getByAltText(article.headlines.basic);
    expect(articleImage).toBeInTheDocument();
  });

  it("renders the article date", () => {
    render(<ArticleComponent article={article} />);

    const articleDate = screen.getByText(formattedDate);
    expect(articleDate).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<ArticleComponent article={article} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
