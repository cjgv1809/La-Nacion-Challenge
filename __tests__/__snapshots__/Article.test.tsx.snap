// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article matches the snapshot 1`] = `
<DocumentFragment>
  <article
    class="mod-caja-nota lugares w-100-mobile"
  >
    <figure
      class="cont-figure"
    >
      <a
        class="figure"
        href="#"
      >
        <picture
          class="content-pic picture"
        >
          <img
            alt="Article headline"
            class="content-img"
            data-nimg="fill"
            decoding="async"
            sizes="
                  (max-width: 640px) 100vw,
                  (max-width: 768px) 50vw,
                  (max-width: 1024px) 33vw,
                  25vw
                "
            src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F800x400&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </picture>
      </a>
    </figure>
    <div
      class="mod-caja-nota__descrip"
    >
      <h2
        class="com-title-acu"
      >
        <a
          href="#"
        >
          <strong>
            Article headline
          </strong>
        </a>
      </h2>
      <p
        class="com-date"
      >
        4 de noviembre de 2024
      </p>
    </div>
  </article>
</DocumentFragment>
`;
